### ISSUE TITLE
Error in sitecustomize ImportError: cannot import name 'LOADED_MODULES' from 'ddtrace' 

### ISSUE THREAD

**Issue by esauser** [DataDog/dd-trace-py#5771]
### Summary of problem
Since `1.12.4` we've ddtrace has failed to initialize with our application.  The application boots and runs just fine, but `ddtrace` is not functioning.  This appears to have started with `1.12.4`.

### Which version of dd-trace-py are you using?
`1.12.4`

### How can we reproduce your problem?
We're seeing the issue when booting Django, Celery Worker, and Celery Beat.

### What is the result that you get?
Notice the last two error lines.
```
datadog autoinstrumentation: installing python package
Looking in links: /datadog-lib/ddtrace_pkgs
Collecting ddtrace
  Downloading ddtrace-1.12.4-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.1 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.1/4.1 MB 60.4 MB/s eta 0:00:00
Collecting ddsketch>=2.0.1
  Downloading ddsketch-2.0.4-py3-none-any.whl (18 kB)
Requirement already satisfied: packaging>=17.1 in /home/user/.local/share/pypoetry/venv/lib/python3.11/site-packages (from ddtrace) (23.1)
Collecting tenacity>=5
  Downloading tenacity-8.2.2-py3-none-any.whl (24 kB)
Collecting cattrs
  Downloading cattrs-22.2.0-py3-none-any.whl (35 kB)
Requirement already satisfied: six>=1.12.0 in /home/user/.local/share/pypoetry/venv/lib/python3.11/site-packages (from ddtrace) (1.16.0)
Collecting typing-extensions
  Downloading typing_extensions-4.5.0-py3-none-any.whl (27 kB)
Requirement already satisfied: jsonschema in /home/user/.local/share/pypoetry/venv/lib/python3.11/site-packages (from ddtrace) (4.17.3)
Collecting xmltodict>=0.12
  Downloading xmltodict-0.13.0-py2.py3-none-any.whl (10.0 kB)
Collecting envier
  Downloading envier-0.4.0-py3-none-any.whl (10 kB)
Requirement already satisfied: attrs>=20 in /home/user/.local/share/pypoetry/venv/lib/python3.11/site-packages (from ddtrace) (22.2.0)
Collecting protobuf>=3
  Downloading protobuf-4.22.4-cp37-abi3-manylinux2014_x86_64.whl (302 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 302.4/302.4 kB 20.9 MB/s eta 0:00:00
Collecting opentelemetry-api>=1
  Downloading opentelemetry_api-1.17.0-py3-none-any.whl (57 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 57.3/57.3 kB 7.6 MB/s eta 0:00:00
Collecting bytecode
  Downloading bytecode-0.14.1-py3-none-any.whl (38 kB)
Collecting deprecated>=1.2.6
  Downloading Deprecated-1.2.13-py2.py3-none-any.whl (9.6 kB)
Collecting importlib-metadata~=6.0.0
  Downloading importlib_metadata-6.0.1-py3-none-any.whl (21 kB)
Requirement already satisfied: setuptools>=16.0 in /home/user/.local/share/pypoetry/venv/lib/python3.11/site-packages (from opentelemetry-api>=1->ddtrace) (65.5.0)
Requirement already satisfied: pyrsistent!=0.17.0,!=0.17.1,!=0.17.2,>=0.14.0 in /home/user/.local/share/pypoetry/venv/lib/python3.11/site-packages (from jsonschema->ddtrace) (0.19.3)
Collecting wrapt<2,>=1.10
  Downloading wrapt-1.15.0-cp311-cp311-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (78 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 78.9/78.9 kB 9.6 MB/s eta 0:00:00
Requirement already satisfied: zipp>=0.5 in /home/user/.local/share/pypoetry/venv/lib/python3.11/site-packages (from importlib-metadata~=6.0.0->opentelemetry-api>=1->ddtrace) (3.15.0)
Installing collected packages: xmltodict, wrapt, typing-extensions, tenacity, protobuf, importlib-metadata, envier, cattrs, bytecode, deprecated, ddsketch, opentelemetry-api, ddtrace
  Attempting uninstall: importlib-metadata
    Found existing installation: importlib-metadata 6.3.0
    Uninstalling importlib-metadata-6.3.0:
      Successfully uninstalled importlib-metadata-6.3.0
Successfully installed bytecode-0.14.1 cattrs-22.2.0 ddsketch-2.0.4 ddtrace-1.12.4 deprecated-1.2.13 envier-0.4.0 importlib-metadata-6.0.1 opentelemetry-api-1.17.0 protobuf-4.22.4 tenacity-8.2.2 typing-extensions-4.5.0 wrapt-1.15.0 xmltodict-0.13.0

[notice] A new release of pip is available: 23.0.1 -> 23.1.2
[notice] To update, run: /home/user/.local/share/pypoetry/venv/bin/python -m pip install --upgrade pip
datadog autoinstrumentation: successfully installed python package
datadog autoinstrumentation: successfully configured python package
Skipping virtualenv creation, as specified in config file.
Error in sitecustomize; set PYTHONVERBOSE for traceback:
ImportError: cannot import name 'LOADED_MODULES' from 'ddtrace' (/usr/local/lib/python3.11/site-packages/ddtrace/__init__.py)
```

I set PYTHONVERBOSE which then yielded this traceback
```
import 'ddtrace' # <_frozen_importlib_external.SourceFileLoader object at 0x7efe7ab7ed10>
Traceback (most recent call last):
  File "<frozen site>", line 548, in execsitecustomize
  File "<frozen importlib._bootstrap>", line 1178, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1149, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/home/user/.local/share/pypoetry/venv/lib/python3.11/site-packages/ddtrace/bootstrap/sitecustomize.py", line 5, in <module>
    from ddtrace import LOADED_MODULES  # isort:skip
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ImportError: cannot import name 'LOADED_MODULES' from 'ddtrace' (/usr/local/lib/python3.11/site-packages/ddtrace/__init__.py)
# destroy sitecustomize
```

### What is the result that you expected?
ddtrace to successfully initialize.

**Comment by ZStriker19** [comment]
Hi @esauser thanks for reaching out. Your issue is most likely caused by this PR that went into 1.12.4. Could you please show me how you're initializing the tracer for your applications?

**Comment by esauser** [comment]
@ZStriker19 I'm not sure what you mean.  We're using the autoinstrumentation so we don't do anything.  It all happens magically.

**Comment by esauser** [comment]
Injection

**Comment by mabdinur** [comment]
Hey @esauser, 

It appears that library injection is installing two versions of the ddtrace library. It appears that one version of the ddtrace library is attempting to import a constant from another version of the ddtrace library. Since `ddtrace.LOADED_MODULES` is only defined in ddtrace v1.12.4 this import fails. 

Why this is occurring is still unclear to me, however this PR ensures ddtrace is only installed from local packages: https://github.com/DataDog/dd-trace-py/pull/5758. This fix should resolve this issue.

**Comment by emmettbutler** [comment]
Because the fix pull request #5758 has been merged, I'm closing this issue. Please reopen if the issue persists.
